### PR TITLE
Segment data based on token

### PR DESCRIPTION
**Pull request by AntonioJanael**
**Describe what this PR introduces:** (if necessary, link currently open issues with [special keywords](https://help.github.com/en/articles/closing-issues-using-keywords))

**What kind of change does this PR introduce?** (check at least one)

- [ ] Bug fix
- [x] Feature
- [ ] Code style update
- [x] Refactor
- [ ] Build-related changes
- [ ] Other, please describe:

**Does this PR introduce any breaking changes?** (check one)

- [ ] Yes
- [x] No

If yes, please describe the impact and migration path for existing applications:

**The PR fulfills these requirements:**

- [ ] Related issues are referenced in the PR (e.g. `fix #xxx[,#xxx]`, where "xxx" is the issue number)
- [x] All tests are passing
- [ ] New/updated tests are included

**Testing environment:**

- Operating System/Platform: Ubuntu 20.04.
- Go version: 1.6.

If you have relevant logs, error output and etc, please attach them.

**Other information:**

### CODE REVIEW

File: pkg/mocks/publisher.go [hunk 1]
```diff
@@ -51,6 +51,11 @@ func (fp *FakePublisher) PublishBroadcastData(thingID, token string, data []enti
 	return args.Error(0)
 }
 
+// PublishSegmentedData provides a mock function to publish data in broadcast mode
```
Comment (rtorresmatheus) [inline pkg/mocks/publisher.go]: Is it same the description of PublishBroadcastData function.

File: pkg/thing/interactors/update_data.go [hunk 2]
```diff
@@ -77,9 +84,9 @@ func validateSchema(data entities.Data, configList []entities.Config) bool {
 // ValidateSchemaNumber validates the value received against its type defined in the sensor's schema
 func ValidateSchemaNumber(value float64, valueType int) bool {
 	switch valueType {
-	case 1: // int
+	case intType:
 		return value >= math.MinInt32 && value <= math.MaxInt32
-	case 2: // float
+	case floatType:
 		return true
 	case 5: // int64
```
Comment (rtorresmatheus) [inline pkg/thing/interactors/update_data.go]: Can you create const for this number too?

**Review by rtorresmatheus — CHANGES_REQUESTED** [review]
Very good! Code is very clean.

File: pkg/mocks/publisher.go (hunk 1 above)
Comment (AntonioJanael) [reply pkg/mocks/publisher.go]: Fixed.

File: pkg/thing/interactors/update_data.go (hunk 2 above)
Comment (AntonioJanael) [reply pkg/thing/interactors/update_data.go]: Yes. Fixed.

**Review by luisfelipemisil — APPROVED** [review]
Very nice and clear. All the changes on the magic numbers are really nice.

**Review by rtorresmatheus — APPROVED** [review]

**Pull request merged**